### DISCUSSION TITLE
More screen uses and shapes

### DISCUSSION THREAD

**Discussion by TIKIRobo** [tsunamayo/Starship-EVO#4040] [Suggestions]
as of rn the screens can only display shield hp, they can technically show the state of a rail only one thing can be hooked into a rail making it useless. 

So what could be done instead is when you press F on the screen a menu appears that would left you select what you want your screen to display (ship & shield HP, fuel and maybe fuel in a specific tank if you hook the screen to that tank/s, heat, speed (this way we can use any screen as a speedometer), and many more!) this way we could have flexibility and the option to display things on the ship itself instead of having to look at/use the hud   

the current shapes of the screens can be a little limited, it would be nice if we could have a bunch of screen shapes to fit any environment

![image](https://user-images.githubusercontent.com/56371294/125837603-0eacfba5-f387-4f7a-9bd7-2d7566b25303.png)
it could be nice to get screens that are more flat and square (but if you feel like screens should have a certain volume you could make them cubes).
![image](https://user-images.githubusercontent.com/56371294/125837747-46298b16-7026-497a-915e-d79db725a154.png)
screens of various sizes and shapes (stretching?) could create an interesting "command" room full of screens displaying every stat about a ship.

**Comment by TIKIRobo** [comment]
something else to think about would be a text screen overhaul, much like the other screens the text screen isn't the easiest to use, my suggestion would be to allow other screens to have text on them and to add more kinds of text codes.

Currently screens can have custom color text as well as **bold** and _italics_ but that isn't at all obvious, the first thing would be to make it more user friendly, either by adding a "cheat sheet" that shows all commands or by making them buttons you can push like on github  
![image](https://user-images.githubusercontent.com/56371294/125838504-21fb919c-2907-4fcb-889c-5926745d5d54.png)
 
![image](https://user-images.githubusercontent.com/56371294/125838551-2d0b3951-9a72-4b0f-86ae-4bd5d78111a9.png)
another thing would be to add more codes, a few builds on the workshop have used text screens to make a sort of hud but these huds are missing key features like indications of current heat, energy or fuel what could be added would be some sort of new text code <heat> would display the ships heat and would update in real time. Since heat numbers can get quite large you could also do <heat/10> which would divide the heat by ten and display that number, typing in <fuel>/<maxfuel> would display the ship's current fuel compared to it's max (100/100)

**Comment by PopkaAkula** [comment]
To build on this more I think stretchy screens might be a really cool thing to have; especially flat and wedge pieces. However I don't really know how feasible it would be though.

**Reply by TIKIRobo** [reply]
stretch is always more "optimal" but a large selection of shapes would work too
